### PR TITLE
chore(ci): Automate publishing to npm

### PR DESCRIPTION
**Pull request by eckdanny-osi**

### CODE REVIEW

**Comment by eckdanny-osi** [comment]
nobody seems to respect the `[skip ci]` string in commit message body 🤔

**Comment by eckdanny-osi** [comment]
@edgibbs ^

File: jenkinsfile.groovy [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
 
-@Field
-def GITHUB_CREDENTIALS_ID = '433ac100-b3c2-4519-b4d6-207c029a103b'
-
 @Field
 def newTag
 
+// Get Secrets
+GITHUB_CREDENTIALS_ID = '433ac100-b3c2-4519-b4d6-207c029a103b'
+NPM_CREDENTIALS_ID = 'MY_NPM_AUTH_TOKEN'
```
Comment (eckdanny-osi) [inline jenkinsfile.groovy]: to whom may i provide?

File: jenkinsfile.groovy [hunk 2]
```diff
@@ -26,20 +32,25 @@ timestamps {
 //
 
 /**
- * Creates pull request pipeline.
+ * Creates pull request pipeline
+ *
+ * The PR pipeline is triggered by GitHub pull request (PR) events.
  *
- * The PR pipeline kicked off by GitHub pull requests (PRs). There is
+ * Note: This does NOT perform an a priori merge with base branch!
```
Comment (eckdanny-osi) [inline jenkinsfile.groovy]: nope

File: jenkinsfile.groovy (hunk 1 above)
Comment (eckdanny-osi) [reply jenkinsfile.groovy]: ed, vinod, oleks...

File: jenkinsfile.groovy [hunk 3]
```diff
@@ -59,104 +70,72 @@ def pullRequestPipeline() {
...
+        stage('Unit Test') {
+          sh "yarn test"
+        }
+        stage('Build Pkgs') {
+          sh "yarn build:libs"
+        }
+        stage('Publish Pkgs') {
+          if (doRelease) {
+            sh "yarn config set '//registry.npmjs.org/:_authToken' ${NPM_CREDENTIALS_ID}"
+            sh "yarn release:publish --yes"
```
Comment (eckdanny-osi) [inline jenkinsfile.groovy]: look at dashboad

File: jenkinsfile.groovy (hunk 1 above)
Comment (edgibbs) [reply jenkinsfile.groovy]: DevOps

**Pull request merged**